lib/qr-generation/qr-generation-screen.dart

```diff
@@ -0,0 +1,113 @@
+import 'package:flutter/material.dart';
+import 'package:qr_flutter/qr_flutter.dart';
+import 'package:qr_generator/models/qr-response-models.dart';
+import 'package:qr_generator/providers/bloc-provider.dart';
+import 'package:qr_generator/qr-generation/qr-generation-bloc.dart';
+
+class QRGenerationScreen extends StatefulWidget {
+  
+  QRGenerationScreen({ Key? key }) : super(key: key);
+
+  @override
+  _QRGenerationScreenState createState() => _QRGenerationScreenState();
+}
+
+class _QRGenerationScreenState extends State<QRGenerationScreen> {
+
+  late QRGenerationBloc bloc;
+
+  @override
+  void initState() {
+    super.initState();
+    bloc = BlocProvider.maybeOf<QRGenerationBloc>(context);
+    bloc.subscribeToGeneratedCode();
+  }
+
+  Widget _buildScannableQRCode(String qrData) {
+    return Expanded(
+      child: Center(
+        child: Padding(
+          padding: EdgeInsets.only(top: 40),
+          child: QrImage(
+            key: Key("qrCode"),
```

Comment:
suggestion: These Keys should probably be made static so that they aren't declared twice between `lib` and `tests`